crates/ty_project/src/db.rs

```diff
@@ -68,24 +68,21 @@ impl ProjectDatabase {
     }
 
     /// Checks all open files in the project and its dependencies.
-    pub fn check(&self) -> Result<Vec<Diagnostic>, Cancelled> {
+    pub fn check(&self) -> Vec<Diagnostic> {
```

Comment:
I decided to remove all `Result`s from here for now because the server already uses many salsa APIs that aren't wrapped in a `salsa::Cancelled::catch`. Instead, I decided that the server request handlers catch salsa cancellations (only necessary for background tasks because:

* Only local tasks can mutate the DB
* All local tasks run on the main loop thread.